src/library/librarycontrol.h

```diff
@@ -65,27 +79,51 @@ class LibraryControl : public QObject {
     void slotDecrementFontSize(double v);
 
   private:
-    Library* m_pLibrary;
-
-    ControlObject* m_pSelectNextTrack;
-    ControlObject* m_pSelectPrevTrack;
-    ControlObject* m_pSelectTrack;
-
-    ControlObject* m_pSelectSidebarItem;
-    ControlObject* m_pSelectPrevSidebarItem;
-    ControlObject* m_pSelectNextSidebarItem;
-
-    ControlObject* m_pToggleSidebarItem;
-    ControlObject* m_pLoadSelectedIntoFirstStopped;
-    ControlObject* m_pAutoDjAddTop;
-    ControlObject* m_pAutoDjAddBottom;
-
-    ControlObject* m_pFontSizeKnob;
-    ControlPushButton* m_pFontSizeIncrement;
-    ControlPushButton* m_pFontSizeDecrement;
-
-    WLibrary* m_pLibraryWidget;
-    WLibrarySidebar* m_pSidebarWidget;
+    Library* m_pLibrary = nullptr;
+
+    // Simulate pressing a key on the keyboard
+    void emitKeyEvent(QKeyEvent&& event);
+
+    // Controls to navigate vertically within currently focussed widget (up/down buttons)
+    ControlPushButton* m_pMoveUp = nullptr;
```

Comment:
Having the creation of the control and connecting its signals next to each other improves understandability -- that's why most controls are created this way instead of being in the member initialization list. I think at least for ControlProxys we could move to a pattern where we create them as member variables (instead of allocating in the constructor) letting the no-args constructor run, and then call ```initialize(ConfigKey)``` alongside the connection calls. It's more complicated for ControlObject derivatives since we pass a lot of configuration through the constructor args.

I think @daschuer's proposed alternative to is to follow the style of most code in Mixxx and not use non-static data member initializers except for the cases he outlined: "For example it might be nice to use it for inline classes, or If there is would be only an empty constructor with an initialisation list."

That sounds reasonable to me. I like using non-static member initialization for classes/structs that are short-and-sweet (e.g. helper structs with no methods). I share the concern that it makes it harder to be sure you've initialized everything and makes you have to check 2 places (or 2 + n for delegating constructors) to figure out the initial value of the class state.